.github/workflows/run_tests_from_a_pr.yml

```diff
@@ -0,0 +1,62 @@
+name: Check running SLOW tests from a PR (only GPU)
+
+on:
+  workflow_dispatch:
+    inputs:
+      docker_image:
+        default: 'diffusers/diffusers-pytorch-cuda'
+        description: 'Name of the Docker image'
+        required: true
+      branch: 
+        description: 'PR Branch to test on'
+        required: true
+      test:
+        description: 'Tests to run (e.g.: `tests/models`).'
+        required: true
+
+env:
+  DIFFUSERS_IS_CI: yes
+  IS_GITHUB_CI: "1"
+  HF_HOME: /mnt/cache
+  OMP_NUM_THREADS: 8
+  MKL_NUM_THREADS: 8
+  PYTEST_TIMEOUT: 600
+  RUN_SLOW: yes
+
+jobs:
+  run_tests:
+    name: "Run a test on our runner from a PR"
+    runs-on: [ self-hosted, intel-cpu, 8-cpu, ci ]
+    container:
+      image: ${{ github.event.inputs.docker_image }}
+      options: --gpus all --privileged --ipc host -v /mnt/cache/.cache/huggingface:/mnt/cache/
+
+    steps:
+      - name: Validate test files input
+        id: validate_test_files
+        run: |
+          if [[ ! "${{ github.event.inputs.test }}" =~ ^[a-zA-Z0-9_/.-\ ]+$ ]]; then
+            echo "Invalid characters detected in test files input"
+            exit 1
+          fi
+          echo ${{ github.event.inputs.test }}
+      
+
+      - name: Checkout PR branch
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.inputs.branch }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
```

Comment:
Does this also work if the PR is created from a fork? I thought it would not, which is why I liked the `gh` approach.

Reply:
I think so, yes. Cc: @glegendre01